src/ert/run_models/ensemble_experiment.py

```diff
@@ -81,23 +81,27 @@ def run_experiment(
 
         self.set_env_key("_ERT_EXPERIMENT_ID", str(self.experiment.id))
         self.set_env_key("_ERT_ENSEMBLE_ID", str(self.ensemble.id))
+        self.set_env_key("_ERT_ITERATION", "0")
+        self.set_env_key("_IS_FINAL_ITERATION", "False")
 
         run_args = create_run_arguments(
             self.run_paths,
             np.array(self.active_realizations, dtype=bool),
             ensemble=self.ensemble,
         )
+
+        self.run_workflows(HookRuntime.PRE_EXPERIMENT, self._storage, self.ensemble)
         sample_prior(
             self.ensemble,
             np.where(self.active_realizations)[0],
             random_seed=self.random_seed,
         )
-
         self._evaluate_and_postprocess(
             run_args,
             self.ensemble,
             evaluator_server_config,
         )
+        self.run_workflows(HookRuntime.POST_EXPERIMENT, self._storage, self.ensemble)
```

Comment:
Should not give storage and ensemble here (they are optional further down in the callstack, so just need to modify the typing)